dio/CHANGELOG.md

```diff
@@ -27,6 +31,8 @@
 - Any classes specific to `dart:io` platforms can now be imported via `import 'package:diox/io.dart';`.
   Classes specific to web can be imported via `import 'package:diox/browser.dart';`.
 - `connectTimeout`, `sendTimeout`, and `receiveTimeout` are now `Duration`s.
+- Interceptor callbacks now take a 3rd argument, the `Dio` instance. This allows to easily perform
+  requests with the same `Dio` instance, for example a token refresh in a `QueuedInterceptor`.
```

Comment:
Is this actually solved the issue? Previously we have https://github.com/flutterchina/dio/issues/1612 and it's linked issues, introducing the previous `Dio` instance doesn't seem to be helpful at this point.

Reply:
Yea maybe not, not sure anymore.
I think `QueuedInterceptor` should be changed to only run the `onRequest` queued, not `onError` and `onResponse`. 
And maybe we can add some interceptor registry with a request matcher, so you can easily configure interceptors to not run for certain requests.
Those things could help with the dead locks but its probably hard to avoid user errors completely.

Reply:
> I think `QueuedInterceptor` should be changed to only run the `onRequest` queued, not `onError` and `onResponse`. And maybe we can add some interceptor registry with a request matcher, so you can easily configure interceptors to not run for certain requests. Those things could help with the dead locks but its probably hard to avoid user errors completely.

Sure. Feel free to open another issue to track ideas.